cpp/tests/utilities/column_wrapper.hpp

```diff
@@ -112,7 +113,10 @@ template <typename Element, typename InputIterator>
 rmm::device_buffer make_elements(InputIterator begin, InputIterator end)
 {
   static_assert(cudf::is_fixed_width<Element>(), "Unexpected non-fixed width type.");
-  thrust::host_vector<Element> elements(begin, end);
+  auto begin_t =
+    thrust::make_transform_iterator(begin, cudf::typecaster<Element>{});  // explicit typecast
```

Comment:
the iterator can produce types that may not convertible to the target type `Element`. hence, you may have to distinguish between cases where the types are convertible and where they do not. for ones that can't be converted, you may have to check on a case by case basis if the iterators value type can be converted to the target type.

please see [here](https://github.com/rapidsai/cudf/blob/8fca5d5073fce281b3cf979fac21694b0747e83b/cpp/tests/utilities/column_wrapper.hpp#L113) and [here](https://github.com/rapidsai/cudf/blob/8fca5d5073fce281b3cf979fac21694b0747e83b/cpp/tests/utilities/column_wrapper.hpp#L157) for one possible approach.

Reply:
Here, explicit non-convertible types won't compile anyway.
explicit and implicit type conversion should have the same generated assembly finally. no need for specialization.

Reply:
there are tests that may create fixed width types with numeric types, which may fail if they aren't convertible. it is being tested today and we want to support that for the new type as well.

Reply:
removed the general iterator explicit typecast. 
Added specialization for initializer_list<int> to chrono types.